pkg/tap/switch.go

```diff
@@ -112,23 +110,10 @@ func (e *Switch) connect(conn net.Conn) (int, bool) {
 }
 
 func (e *Switch) tx(pkt stack.PacketBufferPtr) error {
-	if e.protocol.Stream() {
-		return e.txStream(pkt, e.protocol.(streamProtocol))
-	}
-	return e.txNonStream(pkt)
-}
-
-func (e *Switch) txNonStream(pkt stack.PacketBufferPtr) error {
-	return e.txBuf(pkt, nil)
-}
-
-func (e *Switch) txStream(pkt stack.PacketBufferPtr, sProtocol streamProtocol) error {
-	size := sProtocol.Buf()
-	sProtocol.Write(size, pkt.Size())
-	return e.txBuf(pkt, size)
+	return e.txPkt(pkt)
```

Comment:
For what it's worth, this would have been slightly more straightforward to review if there was a preliminary commit doing these `tx()`  changes while keeping `e.protocol` (a `Move stream socket handling in a txBuf() helper`)
`txPkt` also does not seem very useful in a separate method as `tx` is a one liner, and the only user of this method.
These are just minor commets, this looks good, and worked fine with CRC, so ACK from me.

Reply:
Sure will keep this in mind for the next one 👍